core-api/src/main/java/com/optimizely/ab/bucketing/Bucketer.java

```diff
@@ -115,6 +135,17 @@ private Variation bucketToVariation(@Nonnull Experiment experiment,
             Variation bucketedVariation = experiment.getVariationIdToVariationMap().get(bucketedVariationId);
             logger.info("User \"{}\" is in variation \"{}\" of experiment \"{}\".", userId, bucketedVariation.getKey(),
                         experimentKey);
+
+            // If a persistent bucketer instance is present give it a variation to store
+            if (userExperimentRecord != null) {
+                boolean saved = userExperimentRecord.save(userId, experiment.getKey(), bucketedVariation.getKey());
+                if (saved) {
+                    logger.info("Persisted variation \"{}\" of experiment \"{}\".", bucketedVariation.getKey(), experimentKey);
```

Comment:
put `bucketedVariation.getKey()` in a variable since it's used in multiple places in this method